rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -45,21 +46,28 @@ struct Matcher {
 namespace rir {
 namespace pir {
 
-Value* Rir2Pir::translate() {
+void Rir2Pir::apply(IRCode input) {
+    rir::Function* srcFunction = input.getRirInputFormat()->function;
+    translate(srcFunction, srcFunction->body());
+    //input.pirClosure 
+} 
+
+Value* Rir2Pir::translate(rir::Function* srcFunction, rir::Code* srcCode) {
     assert(!done);
     done = true;
 
-    recoverCFG(srcCode);
+    Builder* insert = cmp.getBuilder();
```

Comment:
`Builder& insert = *cmp.getBuilder()`. Yeah, it looks a bit weird, but it lets us avoid writing `(*insert)(...)` which I think looks weirder.